### PR TITLE
Fixing missing mon[ceph][admin-secret] keyring on radosgw node bnc#930542

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

**Comment by bmwiedemann** [comment]
LGTM

File: chef/cookbooks/ceph/recipes/radosgw.rb [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+
 # needed because of our use of ceph_client
-include_recipe "ceph::keyring"
+ceph_env_filter = " AND ceph_config_environment:ceph-config-default"
+ceph_servers = search(:node, "roles:ceph-osd#{ceph_env_filter}") || []
+if ceph_servers.length > 0
+  include_recipe "ceph::keyring"
+else
+  Chef::Log.info("Ceph was not deployed with Crowbar yet!")
+  return
```
Comment (vuntz) [inline chef/cookbooks/ceph/recipes/radosgw.rb]: Hrm. But that means that after "apply", radosgw will actually not work and we'll have to wait 15 minutes.

I think the right fix would be to not include the keyring recipe if we're a mon, as I believe this is what is creating the issue @bmwiedemann saw.

Comment (tserong) [inline chef/cookbooks/ceph/recipes/radosgw.rb]: Sounds reasonable to me.

Comment (mjura) [inline chef/cookbooks/ceph/recipes/radosgw.rb]: Comment was included

**Comment by mjura** [comment]
@tserong : It is ok Tim, `"ceph::keyring"` is including `"ceph::default"` and `"ceph::conf"` recipes. Just look into https://github.com/crowbar/barclamp-ceph/blob/master/chef/cookbooks/ceph/recipes/keyring.rb#L2

**Comment by mjura** [comment]
PR was tested and works fine!

**Comment by tserong** [comment]
@mjura oh, of course :-) silly me not noticing that.

**Comment by bmwiedemann** [comment]
:+1: this works

**Comment by tserong** [comment]
+1

**Pull request merged**